File: src/main/java/fr/univbrest/dosi/spi/service/ReponseEvaluationService.java
package fr.univbrest.dosi.spi.service;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import fr.univbrest.dosi.spi.bean.ReponseEvaluation;

import fr.univbrest.dosi.spi.dao.ReponseEvaluationRepository;

/**
 * 
 * @author Red1
 *
 */

@Service
public class ReponseEvaluationService {
	
	@Autowired
	ReponseEvaluationRepository ReponseEvaluationRepository;
	
 public ReponseEvaluation	addReponseEvaluation (ReponseEvaluation ReponseEvaluation){
	return ReponseEvaluationRepository.save(ReponseEvaluation);
	 
 }
 

 public final Iterable<ReponseEvaluation> getAllReponseEvaluation() {
		final Iterable<ReponseEvaluation> reponses = ReponseEvaluationRepository.findAll();
		return reponses;
	}
 
 
 
 public void deleteReponseEvaluation(long idReponseEvaluation)
	{
			ReponseEvaluationRepository.delete(idReponseEvaluation);
	}
 
 
 public ReponseEvaluation updateReponseEvaluation(ReponseEvaluation ReponseEvaluation)
	{
	 return ReponseEvaluationRepository.save(ReponseEvaluation);
	}
 
 	
 public ReponseEvaluation getReponseEvaluation(long idReponseEvaluation ){

		return ReponseEvaluationRepository.findOne(idReponseEvaluation);
 		}
 
	}
 	

File: src/main/resources/client/scripts/reponseEvaluation/reponseevaluationservice.js


angular.module('app')

    .factory('ReponseEvaluationService', ['$http','$location', function ($http,$location) {


        var urlBase = 'http://localhost:8090/reponseEvaluation';
        var dataFactory = {};

        dataFactory.getAll = function () {
            return $http.get(urlBase);
        };

        dataFactory.getreponseEvaluation = function (id) {
            return $http.get(urlBase + '/' + id);
        };

        dataFactory.getreponseEvaluation = function (id,noetudiant) {
            return $http.get(urlBase + '/' + id +'/'+noetudiant);
        };

        dataFactory.addreponseEvaluation = function (entity) {
            return $http.post(urlBase , entity);

        };
        
        dataFactory.updatereponseEvaluation = function (entity) {
            return $http.put(urlBase, entity)
        };
        dataFactory.deletereponseEvaluation = function (id) {
            return $http.delete(urlBase+'/'+"delete"+'/'+id, entity)
        };
    }]);


